package.json

```diff
@@ -39,9 +39,11 @@
     "build:prod": "npm-run-all build:prod:frontend",
     "build:prod:frontend": "lerna run build:prod:frontend",
     "build:dev": "npm-run-all build:dev:backend webpack:dev:web build:dev:frontend",
-    "build:dev:backend": "npm-run-all lint clean codegen tsc",
+    "build:dev:backend": "npm-run-all lint clean codegen build:dev:backend:prebuild tsc build:dev:backend:postbuild",
     "build:dev:frontend": "lerna run build:dev:frontend --scope=\"@hyperledger/cactus-example-supply-chain-frontend\" && lerna run build:dev:frontend --scope=\"@hyperledger/cactus-example-carbon-accounting-frontend\"",
     "build:dev:common": "lerna exec --stream --scope '*/*common' -- 'del-cli dist/** && tsc --project ./tsconfig.json && webpack --env=dev --target=node --config ../../webpack.config.js'",
+    "build:dev:backend:prebuild": "lerna run build:dev:backend:prebuild",
+    "build:dev:backend:postbuild": "lerna run build:dev:backend:postbuild",
```

Comment:
@petermetz 
Projects changed in this PR needs some logic executed before and after it's build (to copy specific files in most cases). This can be simplified in the future, especially prebuild step could be removed I suppose, but for now I think adding pre/post build steps is the most clean solution. On the other hand, maybe it would be reasonable to replace plain `tsc` call with some lerna target present in all projects that would build the project? It's possible that in the future other projects may have more complicated build process, and this way we'd encapsulate all build logic in given package "build" npm script, instead of relying that they all will just work with `tsc`? What's your opinion on this?

Reply:
@outSH Been there, done that. :-) I called the script `resources` at the time: https://github.com/hyperledger/cactus/blob/0a13eff5b1bc3b8f45960d6ca59daf155ae446a0/package.json#L24 
back in https://github.com/hyperledger/cactus/pull/719
You are doing the same thing that I was back then, just with a different name for the post build script. (Which I'm fine with, whatever. My reasoning for using shorter script names where possible was that they are easier to type out when you are running them manually from a terminal but I also see the benefit if fully qualified names like the ones you gave them)

For the prebuild script, you could just use the `codegen` script instead of introducing a new one called `build:dev:backend:prebuild` which would have the performance benefit of running less lerna invocations (which are project wide). Even if the expression `codegen` does not exactly match what you'll end up doing in it inside the customized packages (but IMO it kinda does too) I think it would be descriptive enough compared to something generic like pre build.

Making `tsc` a lerna target could turn out to be a rabbit hole because whatever we do in there needs to play nice with the watch scripts and the IDE's own processes that powers auto-complete, syntax highlighting/etc. => 
The `tsc` script used to be a lerna target, but that meant that you had to exit and launch VSCode again every time you re-built the project because it would get out of sync and all the imports would go red in VSCode even though on the terminal the compiler was giving you no errors. I'm not saying that this is a reason to not to do it/look into it, just that there's a history there which we'd need to avoid repeating because the developer experience was quite bad then (not that it couldn't be improved now, but still even worse)

TLDR: I'm fine with this, up to you if you want to refactor it based on the above information.

Reply:
@petermetz Thank you very much for such a detailed explanation :)

> For the prebuild script, you could just use the codegen script instead of introducing a new one called `build:dev:backend:prebuild`

Agree, changed `prebuild` to `codegen`, didn't notice this before. I've left postbuild as is, I don't think this should be called in separation from the whole build, can be changed to something simpler in the future if I'm wrong.

Reply:
@outSH You are welcome, glad it was at least somewhat helpful. 

> Agree, changed prebuild to codegen, didn't notice this before. I've left postbuild as is, I don't think this should be called in separation from the whole build, can be changed to something simpler in the future if I'm wrong.

Sounds good to me.